snippets/paging-cursor.js

```diff
@@ -0,0 +1,48 @@
+// the perform method of our trigger
+// ensure operation.canPaginate is true!
+
+const performWihoutAsync = (z, bundle) => {
+  return Promise.resolve()
+    .then(() => {
+      if (bundle.meta.page === 0) {
+        // first page, no need to fetch a cursor
+        return Promise.resolve();
+      } else {
+        return z.cursor.get(); // Promise<string | null>
+      }
+    })
+    .then(cursor => {
+      return z.request(
+        'https://5ae7ad3547436a00143e104d.mockapi.io/api/recipes',
+        {
+          params: { cursor: cursor } // if cursor is null, it's ignored here
+        }
+      );
+    })
+    .then(response => {
+      // need to save the cursor and return a promise, but also need to pass the data along
+      return Promise.all([response.items, z.cursor.set(response.nextPage)]);
+    })
+    .then(promises => {
+      // [items[], null]
```

Comment:
I'd go ahead and use destructuring here unless we're intentionally not doing that to try to keep the docs really simple

```js
.then(([items, /* null */]) => {
  return items;
});
```